### PR TITLE
Module 7 Routing

### PR DESCRIPTION
**Pull request by TashaUA**

### CODE REVIEW

File: src/app/modules/components/auth/auth.component.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+})
+export class AuthComponent implements OnInit {
+
+  constructor(private authService: AuthService, private router: Router,) { }
+
+  ngOnInit(): void {
+  }
+
+  login(e: any) {
+    e.preventDefault();
```
Comment (goremchik) [inline src/app/modules/components/auth/auth.component.ts]: For click event, no need to add prevent default (only for submit)

File: src/app/modules/components/courses/add-edit-form/add-edit-form.component.ts [hunk 2]
```diff
@@ -13,12 +15,16 @@ import { FormBuilder } from '@angular/forms';
...
-    private formBuilder: FormBuilder) { }
+    private formBuilder: FormBuilder,
+    public dialogRef: MatDialogRef<AddEditPageComponent>,
+    ) {
+    console.log(this.courseToEditId, this.coursesService.getCourseById(this.courseToEditId));
+  }
 
-  public course: Course = this.courseToEdit || {title: '', description: '', creationDate: '', duration: 0};
-  public isEdit: Boolean = !!this.courseToEdit;
+  public course: Course = this.coursesService.getCourseById(this.courseToEditId) || {id: 0, title: '', description: '', creationDate: '', duration: 0};
```
Comment (goremchik) [inline src/app/modules/components/courses/add-edit-form/add-edit-form.component.ts]: It is not quite good practice to define such empty object as default value. it is much better to set null

File: src/app/modules/components/courses/add-edit-form/add-edit-form.component.ts [hunk 3]
```diff
@@ -13,12 +15,16 @@ import { FormBuilder } from '@angular/forms';
...
+  public isEdit: Boolean = !!this.courseToEditId;
```
Comment (goremchik) [inline src/app/modules/components/courses/add-edit-form/add-edit-form.component.ts]: `: boolean` (not `: Boolean`)

File: src/app/pages/auth-page/auth-page.component.ts [hunk 4]
```diff
@@ -8,9 +12,22 @@ import {ChangeDetectionStrategy, Component, OnInit} from '@angular/core';
 })
 export class AuthPageComponent implements OnInit {
 
-  constructor() { }
+  constructor( public dialog: MatDialog, private router: Router) {    this.openModal(); }
```
Comment (goremchik) [inline src/app/pages/auth-page/auth-page.component.ts]: Please add all logic to ngOnInit

File: src/app/modules/components/auth/auth.component.ts (hunk 1 above)
Comment (TashaUA) [reply src/app/modules/components/auth/auth.component.ts]: done

File: src/app/modules/components/courses/add-edit-form/add-edit-form.component.ts (hunk 2 above)
Comment (TashaUA) [reply src/app/modules/components/courses/add-edit-form/add-edit-form.component.ts]: fixed

File: src/app/modules/components/courses/add-edit-form/add-edit-form.component.ts (hunk 3 above)
Comment (TashaUA) [reply src/app/modules/components/courses/add-edit-form/add-edit-form.component.ts]: done

**Pull request merged**